src/main/java/org/springframework/data/neo4j/core/mapping/CypherGenerator.java

```diff
@@ -82,6 +84,33 @@ public enum CypherGenerator {
 
 	INSTANCE;
 
+	// keeping elementId/id function selection in one place within this class
+	// default elementId function
+	private Function<Named, FunctionInvocation> elementIdOrIdFunction = named -> {
+		if (named instanceof Node node) {
+			return Functions.elementId(node);
+		} else if (named instanceof Relationship relationship) {
+			return Functions.elementId(relationship);
+		} else {
+			throw new IllegalArgumentException("Unsupported CypherDSL type: " + named.getClass());
+		}
+	};
+
+	/**
+	 * Set elementIdFunction to be used for identifier function.
+	 *
+	 * @param newElementIdOrIdFunction elementId CypherDSL function
+	 * @return the modified CypherGenerator
+	 */
+	public CypherGenerator withElementIdFunction(Function<Named, FunctionInvocation> newElementIdOrIdFunction) {
```

Comment:
Why the modified one and not a new one? If you modify it (which is ok for me), use a setter.
Also, you can leave the instance in the mapping context final if you modify it anyway.